File: SDKs/ote-godaddy.com/aftermarket/1.0.0/python/src/sdk/models/operations/deletelistings.py
import dataclasses
from typing import Any,Optional


@dataclasses.dataclass
class DeleteListingsQueryParams:
    domains: list[str] = dataclasses.field(metadata={'query_param': { 'field_name': 'domains', 'style': 'form', 'explode': False }})
    

@dataclasses.dataclass
class DeleteListingsRequest:
    query_params: DeleteListingsQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class DeleteListingsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    aftermarket_listing_action: Optional[Any] = dataclasses.field(default=None)
    body: Optional[bytes] = dataclasses.field(default=None)
    error: Optional[Any] = dataclasses.field(default=None)
    error_limit: Optional[Any] = dataclasses.field(default=None)
    

File: SDKs/ote-godaddy.com/aftermarket/1.0.0/typescript/src/sdk/models/operations/deletelistings.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class DeleteListingsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=domains" })
  domains: string[];
}


export class DeleteListingsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: DeleteListingsQueryParams;
}


export class DeleteListingsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  aftermarketListingAction?: any;

  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: any;

  @SpeakeasyMetadata()
  errorLimit?: any;

  @SpeakeasyMetadata()
  statusCode: number;
}
